web/src/main/java/com/microsoft/applicationinsights/web/internal/WebRequestTrackingFilter.java

```diff
@@ -59,20 +58,24 @@
     private boolean agentIsUp = false;
     private final LinkedList<ThreadLocalCleaner> cleaners = new LinkedList<ThreadLocalCleaner>();
 
+    //Only for Spring Boot Application (External Name Setting)
```

Comment:
1. As soon as we make new constructor public, we cannot control if it's going to be used for Spring Boot only or not.
2. There might be another framework we haven't tried yet that required application name setup outside of configuration xml files.